### PR TITLE
Modularise the bot code

### LINKED ISSUE #75: [Suggestion] Don't cancel loans that can't be replaced
Sometimes, loans get partially filled and leave something like .0009 leftover, then the bot comes and cancels it.

Since the balance isn't over .001, the bot can't replace the loan, wasting time and money.

Solution: Don't cancel loans under .001

### LINKED ISSUE #131: Bot may try to lend under Poloniex minimum
It is possible to use the coincfg to configure the bot to place a lend at a value lower than 0.0001, which gets rejected by Poloniex and throws an error.

Solution: check rate before placing lend order.

### PR DESCRIPTION
**Pull request by Evanito**
## Description

Break down a lot of the bot's code into more maintainable chunks. This also made it easier to do a lot of other things! 🚀 

Solves #138 
Fixes #131 
Fixes #75

It is important to address the problems caused by having a high churn/high complexity file. These files become bottlenecks of maintainability and readability as they only get more complicated, this PR is an answer to how our Repo is suffering from a black hole of code in lendingbot.py

I am currently adding more as I go on, I just want to make this PR so Codacy can evaluate it and I can get some feedback.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] **I have read CONTRIBUTING.md**
- [x] **I fully understand [Github Flow.](https://guides.github.com/introduction/flow/)**
- [x] **My code adheres to the [code style of this project.](https://poloniexlendingbot.readthedocs.io/en/latest/contributing.html)**
- [x] **I have updated the documentation in /docs if I have changed the config, arguments, logic in how the bot works, or anything that understandably needs a documentation change.**
- [x] **I have updated the config file accordingly if my change requires a new configuration setting or changes an existing one.**
- [x] **I have tested the bot with no issues for 24 continuous hours. If issues were experienced, they have been patched and tested again.**

### CODE REVIEW

**Comment by Evanito** [comment]
Guess what I disabled again! 👎

File: modules/Configuration.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+def has_option(category, option):
+    return config.has_option(category, option)
+
+
+def getboolean(category, option):
+    return config.getboolean(category, option)
+
+
+def get(category, option, default_value=False, lower_limit=False, upper_limit=False):
```
Comment (Evanito) [inline modules/Configuration.py]: I mentioned that usage in a comment on lendingbot.py, but I definitely intend to update the  docs with this iinformation.

Comment (Evanito) [inline modules/Configuration.py]: Done

**Comment by Evanito** [comment]
Okay, I think I'm ready to call this complete.

Please review @rnevet

**Comment by rnevet** [comment]
lgtm

**Comment by rnevet** [comment]
minor comment: 
KeepStuckOrders - new config parameter is CamelCased while most others aren't.

**Comment by rnevet** [comment]
I think we can optimize the amount_to_lend function also.

File: docs/configuration.rst [hunk 2]
```diff
@@ -139,34 +146,34 @@ Very few situations require you to change these settings.
 
 This feature group allows you to only lend a certain percentage of your total holding in a coin, until the lending rate suprasses a certain threshhold. Then it will lend at max capacity.
 
-``maxtolent`` is a raw number of how much you will lend of each coin whose lending rate is below ``maxtolentrate``.
+``maxtolend`` is a raw number of how much you will lend of each coin whose lending rate is below ``maxtolendrate``.
```
Comment (rnevet) [inline docs/configuration.rst]: I think you wanted to change this parameter name to lend, but did this only in the documentation.

Comment (Evanito) [inline docs/configuration.rst]: Doh. I will get to this after we resolve #143

**Comment by Evanito** [comment]
@rnevet KeepStuckOrders -> keepstuckorders
done

**Comment by rnevet** [comment]
Lgtm

**Pull request merged**